.github/workflows/ci.yml

```diff
@@ -14,14 +14,14 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: set up python
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v5
```

Comment:
At the risk of being quite annoying here, should we just use `uv` here and simplify things by removing the python setups, etc?

Reply:
Maybe not worth the effort for v1, although you've already invested a lot here...

Reply:
Yes best to keep things as is, or we should move away completely from `setuptools` and use `uv` as well as a build backend which is a bigger task considering we need support for the Cython build extension.